web/app/scripts/navbar/navbar-controller.js

```diff
@@ -2,45 +2,24 @@
     'use strict';
 
     /* ngInject */
-    function NavbarController($log, $rootScope, $state, $stateParams, RecordTypes) {
+    function NavbarController($log, $rootScope, $state, $stateParams,
```

Comment:
I wonder if there might be benefit in making a service to handle the responsibility for tracking the selected and available Geographies, Polygons, and RecordTypes. That would potentially simplify `setResourceObjects` because you would be able to remove the logic that sets the controller values via the `$stateParams` since a service would be less transient. It would also make things easier for other parts of the app that might need to respond to changes in the selected types, which several of them might.